### PR TITLE
Add kubernetes manifests for Heartbeat

### LINKED ISSUE #17752: Add kubernetes manifests for Heartbeat
**Describe the enhancement:**

We don't currently offer Kubernetes manifests to deploy Heartbeat. This issues is minded to create and document manifests for Heartbeat like we do for others here: https://github.com/elastic/beats/tree/master/deploy/kubernetes

### PR DESCRIPTION
**Pull request by exekias**
## What does this PR do?

Add Kubernetes example manifests for Heartbeat, with some examples on how to use autodiscover there.

## Related issues

Closes elastic/beats#17752

### CODE REVIEW

File: heartbeat/docs/running-on-kubernetes.asciidoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+curl -L -O https://raw.githubusercontent.com/elastic/beats/{branch}/deploy/kubernetes/{beatname_lc}-kubernetes.yaml
+------------------------------------------------
+
+[WARNING]
+=======================================
+If you are using Kubernetes 1.7 or earlier: {beatname_uc} uses a hostPath volume to persist internal data, it's located
+under /var/lib/{beatname_lc}-data. The manifest uses folder autocreation (`DirectoryOrCreate`), which was introduced in
```
Comment (ChrsMark) [inline heartbeat/docs/running-on-kubernetes.asciidoc:33]: It seems that there is no `DirectoryOrCreate` volume in the manifests.

Comment (exekias) [reply heartbeat/docs/running-on-kubernetes.asciidoc:33]: Good catch! 0db39e7

Comment (ChrsMark) [reply heartbeat/docs/running-on-kubernetes.asciidoc:33]: 👍

**Comment by exekias** [comment]
@andrewvc PTAL, happy to go over these changes together

**Pull request merged**